### PR TITLE
Server 513. Closes #513.

### LINKED ISSUE #513: Throw specific error for Token expired
## Description
As a client I need to know when a token  is expired

## Acceptance criteria

- [x] New error type for Auth Token expired
- [x] Log a warning if the env var AAD_NO_PII is set to true as this is blocked by a M$ issue and it is always set to false

### PR DESCRIPTION
**Pull request by valentinyanakiev**
### Describe the background of your pull request

Returns TokenException with CherrytwistErrorStatus.TOKEN_EXPIRED when JWT token has expired.

### Additional context

Works only with AUTH_AAD_LOGGING_PII=true. Microsoft has decided that verbose validation logging needs PII. Go figure.
The culprit is on line 330 in the passport-azure-ad/lib/bearerstrategy.js (you can find it in the node_modules after npm install).

    if (err) {
      if (err.message && !self._options.loggingNoPII)
        return self.failWithLog(err.message);
      else
        return self.failWithLog('In Strategy.prototype.jwtVerify: cannot verify token');
    }

There is a bunch of code similar to this one in the library.

### Governance 

- [x] I've read the contributing document https://github.com/cherrytwist/.github/blob/master/CONTRIBUTING.md
- [x] I've read and understand the Code of Conduct https://github.com/cherrytwist/.github/blob/master/CODE_OF_CONDUCT.md
- [x] I understand that any contributions or suggestions I made may make it into the actual code. I've read the License 
     https://github.com/cherrytwist/Coordination/blob/master/LICENSE 
     and the contributor license agreement: tba

### CODE REVIEW

**Review by techsmyth — APPROVED** [review]
Well found...

File: src/utils/bootstrap/bootstrap.service.ts [hunk 1]
```diff
@@ -84,6 +84,15 @@ export class BootstrapService {
         );
       const entries = Object.entries(config);
       for (const [key, value] of entries) {
+        if (
+          key === 'loggingNoPII' &&
+          value === false &&
+          process.env.AUTH_AAD_LOGGING_PII === 'false'
```
Comment (techsmyth) [inline src/utils/bootstrap/bootstrap.service.ts:90]: is it not that the case we are overriding is if the env var is set to true? Also looks like code would still set the value...

Comment (valentinyanakiev) [reply src/utils/bootstrap/bootstrap.service.ts:90]: The env var is true when we are logging PII, and the key is true when no PII is logged. It's inversed logic.

Comment (valentinyanakiev) [reply src/utils/bootstrap/bootstrap.service.ts:90]: loggingNoPII has the opposite value of  process.env.AUTH_AAD_LOGGING_PII - that's what we check. Whether the hardcoded value is different than the env variable.

**Pull request merged**